src/main/kotlin/pt/up/fe/ni/website/backend/util/FileUploader.kt

```diff
@@ -0,0 +1,47 @@
+package pt.up.fe.ni.website.backend.util
+
+import com.cloudinary.Cloudinary
+import com.cloudinary.Transformation
+import java.io.File
+
+interface FileUploader {
+    fun upload(folder: String, fileName: String, image: ByteArray): String
+    fun delete(filePath: String)
+}
+
+class CloudinaryFileUploader(private val basePath: String, private val cloudinary: Cloudinary) : FileUploader {
+    override fun upload(folder: String, fileName: String, image: ByteArray): String {
+        val path = "$basePath/$folder/$fileName"
+
+        val imageTransformation = Transformation().width(250).height(250).crop("thumb").chain()
+
+        val result = cloudinary.uploader().upload(
+            image,
+            mapOf(
+                "public_id" to path,
+                "overwrite" to true,
+                "transformation" to imageTransformation,
+            )
+        )
+
+        return result["url"]?.toString() ?: ""
+    }
+
+    override fun delete(filePath: String) {
+        println("upload cloud")
```

Comment:
Is this for a future issue? We can use the TODO exception for that

Reply:
It's cool to have but not a priority I think. I would leave but I think we should throw the exception, a print might not be readable enough in the logs

Reply:
It's probably better to simply remove this entirely and open a new issue